client/my-sites/plans-features-main/hooks/use-is-plan-upsell-enabled-on-free-domain.ts

```diff
@@ -0,0 +1,44 @@
+import { useExperiment } from 'calypso/lib/explat';
+import { useSelector } from 'calypso/state';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
+
+export const useIsPlanUpsellEnabledOnFreeDomain = (
+	flowName?: string | null,
+	freeSubDomainName?: string
+): DataResponse< boolean > => {
+	const userLocale = useSelector( getCurrentUserLocale );
+
+	const ONBOARDING_EXPERIMENT = 'calypso_gf_signup_onboarding_free_free_dont_miss_out_modal';
+	const ONBOARDING_PM_EXPERIMENT = 'calypso_gf_signup_onboarding_pm_free_free_dont_miss_out_modal';
+
+	const [ isLoadingPaidMediaAssignment, paidMediaAssignment ] = useExperiment(
+		ONBOARDING_EXPERIMENT,
+		{
+			isEligible: userLocale === 'en' && flowName === 'onboarding' && !! freeSubDomainName,
+		}
+	);
+
+	const [ isLoadingOnboardingAssignment, onboardingAssignment ] = useExperiment(
+		ONBOARDING_PM_EXPERIMENT,
+		{
+			isEligible: userLocale === 'en' && flowName === 'onboarding-pm' && !! freeSubDomainName,
+		}
+	);
+
+	if ( userLocale !== 'en' ) {
+		return { isLoading: false, result: false };
+	}
+
+	return {
```

Comment:
Following the above, we actually don't have to add these conditions here since the ExPlat hook handles that already. We just need to return the value accordingly here:


```
isLoading: isLoadingPaidMediaAssignment,
result: experimentAssignment?.variationName === 'treatment'
```